src/application/engine/game_object/Body.java

```diff
@@ -0,0 +1,175 @@
+     * @param type The type of body, i.e. rectangular or circle etc.
+     */
+    public Body(GameObject gameObject, Point2D position, double width, double height, Geometry type)
+    {
+        this.type = type;
+        this.gameObject = gameObject;
+        this.position = position;
+        this.width = width;
+        this.height = height;
+        this.orientation = new Vector2(0, 0);
+    }
+
+    /**
+     * The position of the top left corner of this body.
+     * @return Returns the position of the top left corner of this body.
+     */
+    public Point2D getPosition()
+    {
+        return position;
+    }
+
+    /**
+     * Sets the center position of this body.
+     * @param position The new center position of this body.
+     */
+    public void setCenter(Point2D position)
+    {
+        this.position = new Point2D(position.getX() - width / 2, position.getY() - height / 2);
+    }
+
+    /**
+     * Sets the top left corner position of this body.
+     * @param position The new top left corner position of this body.
+     */
+    public void setPosition(Point2D position)
+    {
+        this.position = position;
+    }
+
+    /**
+     * The center position of this body.
+     * @return Returns the current center position of this body.
+     */
+    public Point2D getCenter()
+    {
+        return new Point2D((position.getX() + (width / 2)), position.getY() + (height / 2));
+    }
+
+    /**
+     * The height of this body.
+     * @return Returns the current height of this body.
+     */
+    public double getHeight()
+    {
+        return height;
+    }
+
+    /**
+     * Sets the height of this body
+     * @param height The new height of this body. 
+     */
+    public void setHeight(double height)
+    {
+        this.height = height;
+    }
+
+    /**
+     * The width of this body.
+     * @return Returns the current width of this body.
+     */
+    public double getWidth()
+    {
+        return width;
+    }
+    
+    /**
+     * Sets the width of this body.
+     * @param width The new width of this body.
+     */
+    public void setWidth(double width)
+    {
+        this.width = width;
+    }
+
+    /**
+     * The orientation of this body described as a vector.
+     * @return Returns the orientation of this body as a vector.
+     */
+    public Vector2 getOrientation()
+    {
+        return orientation;
+    }
+
+    /**
+     * Sets the orientation of this body as a vector.
+     * @param orientation The new vector to describe the orientation of this boy.
+     */
+    public void setOrientation(Vector2 orientation)
+    {
+        this.orientation = orientation;
+    }
+
+    /**
+     * Increases the current position of this body.
+     * @param xIncrease The amount to add to the current x value of the body.
+     * @param yIncrease The amount to add to the current y value of the body.
+     */
+    public void increasePosition(double xIncrease, double yIncrease)
```

Comment:
i still would like to see the method name changed as well as the parameters names